pkg/admission/zpdb_validator.go

```diff
@@ -0,0 +1,97 @@
+package admission
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/spanlogger"
+	v1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/grafana/rollout-operator/pkg/zpdb"
+)
+
+const (
+	ZpdbValidatorWebhookPath = "/admission/zpdb-validation"
+)
+
+type zpdbValidatingHandler struct {
+	ctx     context.Context
+	logger  *spanlogger.SpanLogger
+	request v1.AdmissionReview
+}
+
+func (v *zpdbValidatingHandler) initLogger() {
+	v.logger.SetSpanAndLogTag("object.name", v.request.Request.Name)
+	v.logger.SetSpanAndLogTag("object.resource", v.request.Request.Resource.Resource)
+	v.logger.SetSpanAndLogTag("object.namespace", v.request.Request.Namespace)
+	// note that this is the UID of the request, not of the pod
```

Comment:
I'm not sure I follow this comment - what pod could this be referring to when we're validating a ZPDB object?